app/views/provider_interface/application_choices/_application_choice_header.html.erb

```diff
@@ -44,6 +44,22 @@
       </div>
     </div>
   </div>
+<% elsif @rejection_reason_required && flash.empty? && @provider_can_respond -%>
```

Comment:
Not necessarily in this PR but it looks like `_application_choice_header.html.erb` is getting complex enough to be a component or have some backing library to help with conditionals. The instance var setup in the controller could then be reduced.